// r6statapi/dist/esm/interfaces/marketplace.js
export var ItemRarity;
(function (ItemRarity) {
    ItemRarity["Legendary"] = "rarity_legendary";
    ItemRarity["SuperRare"] = "rarity_superrare";
    ItemRarity["Rare"] = "rarity_rare";
    ItemRarity["Uncommon"] = "rarity_uncommon";
})(ItemRarity || (ItemRarity = {}));
// | 'Y9S4';
export var Operators;
(function (Operators) {
    Operators["Ace"] = "Character.Y5S2.ACE";
    Operators["Alibi"] = "Character.Y3S2.ALIBI";
    Operators["Amaru"] = "Character.Y4S3.AMARU";
    Operators["Aruni"] = "Character.Y5S4.ARUNI";
    Operators["Ash"] = "Character.Legacy.ASH";
    Operators["Azami"] = "Character.Y7S1.AZAMI";
    Operators["Bandit"] = "Character.Legacy.BANDIT";
    Operators["Blackbeard"] = "Character.Legacy.BLACKBEARD";
    Operators["Blitz"] = "Character.Legacy.BLITZ";
    Operators["Brava"] = "Character.Y8S1.BRAVA";
    Operators["Buck"] = "Character.Legacy.BUCK";
    Operators["Capitao"] = "Character.Legacy.CAPITAO";
    Operators["Castle"] = "Character.Legacy.CASTLE";
    Operators["Caveria"] = "Character.Legacy.CAVEIRA";
    Operators["Clash"] = "Character.Y3S3.CLASH";
    Operators["Deimos"] = "Character.Y9S1.DEIMOS";
    Operators["Doc"] = "Character.Legacy.DOC";
    Operators["Dokkaebi"] = "Character.Y2S4.DOKKAEBI";
    Operators["Echo"] = "Character.Legacy.ECHO";
    Operators["Ela"] = "Character.Y2S3.ELA";
    Operators["Fenrir"] = "Character.Y8S2.FENRIR";
    Operators["Finka"] = "Character.Y3S1.FINKA";
    Operators["Flores"] = "Character.Y6S1.FLORES";
    Operators["Frost"] = "Character.Legacy.FROST";
    Operators["Fuze"] = "Character.Legacy.FUZE";
    Operators["Glaz"] = "Character.Legacy.GLAZ";
    Operators["Goyo"] = "Character.Y4S3.GOYO";
    Operators["Gridlock"] = "Character.Y4S1.GRIDLOCK";
    Operators["Grim"] = "Character.Y7S3.GRIM";
    Operators["Hibana"] = "Character.Legacy.HIBANA";
    Operators["Iana"] = "Character.Y5S1.IANA";
    Operators["IQ"] = "Character.Legacy.IQ";
    Operators["Jackal"] = "Character.Y2S1.JACKAL";
    Operators["Jager"] = "Character.Legacy.JAGER";
    Operators["Kaid"] = "Character.Y3S4.KAID";
    Operators["Kali"] = "Character.Y4S4.KALI";
    Operators["Kapkan"] = "Character.Legacy.KAPKAN";
    Operators["Lesion"] = "Character.Y2S3.LESION";
    Operators["Lion"] = "Character.Y3S1.LION";
    Operators["Maestro"] = "Character.Y3S2.MAESTRO";
    Operators["Maverick"] = "Character.Y3S3.MAVERICK";
    Operators["Melusi"] = "Character.Y5S2.MELUSI";
    Operators["Mira"] = "Character.Y2S1.MIRA";
    Operators["Montagne"] = "Character.Legacy.MONTAGNE";
    Operators["Mozzie"] = "Character.Y4S1.MOZZIE";
    Operators["Mute"] = "Character.Legacy.MUTE";
    Operators["Nokk"] = "Character.Y4S2.NOKK";
    Operators["Nomad"] = "Character.Y3S4.NOMAD";
    Operators["Oryx"] = "Character.Y5S1.ORYX";
    Operators["Osa"] = "Character.Y6S3.OSA";
    Operators["Pulse"] = "Character.Legacy.PULSE";
    Operators["Ram"] = "Character.Y8S3.RAM";
    Operators["Rook"] = "Character.Legacy.ROOK";
    Operators["Sens"] = "Character.Y7S2.SENS";
    Operators["Sentry"] = "Character.Y9S2.SENTRY";
    Operators["Skopos"] = "Character.Y9S3.SKOPOS";
    Operators["Sledge"] = "Character.Legacy.SLEDGE";
    Operators["Smoke"] = "Character.Legacy.SMOKE";
    Operators["Solis"] = "Character.Y7S4.SOLIS";
    Operators["Striker"] = "Character.Y9S2.STRIKER";
    Operators["Tachanka"] = "Character.Legacy.TACHANKA";
    Operators["Thatcher"] = "Character.Legacy.THATCHER";
    Operators["Thermite"] = "Character.Legacy.THERMITE";
    Operators["Thorn"] = "Character.Y6S4.THORN";
    Operators["Thunderbird"] = "Character.Y6S2.THUNDERBIRD";
    Operators["Tubarao"] = "Character.Y8S4.TUBARAO";
    Operators["Twitch"] = "Character.Legacy.TWITCH";
    Operators["Valkrie"] = "Character.Legacy.VALKYRIE";
    Operators["Vigil"] = "Character.Y2S4.VIGIL";
    Operators["Wamai"] = "Character.Y4S4.WAMAI";
    Operators["Warden"] = "Character.Y4S2.WARDEN";
    Operators["Ying"] = "Character.Y2S3.YING";
    Operators["Zero"] = "Character.Y5S3.ZERO";
    Operators["Zofia"] = "Character.Y2S4.ZOFIA";
})(Operators || (Operators = {}));
export var Weapons;
(function (Weapons) {
    Weapons[".44 Mag Semi-Auto"] = "44_Mag_Semi-Auto";
    Weapons["5.7 USG"] = "57_USG";
    Weapons["6P41"] = "6P41";
    Weapons["9mm C1"] = "9mm_C1";
    Weapons["9x19VSN"] = "9x19VS";
    Weapons["416-C CARBINE"] = "416-C_CARBINE";
    Weapons["FourOneSeven"] = "417";
    Weapons["552 COMMANDO"] = "552_COMMANDO";
    Weapons["556XL"] = "556XL";
    Weapons["1911 TACOPS"] = "1911_TACOPS";
    Weapons["ACS12"] = "ACS12";
    Weapons["AK-12"] = "AK-12";
    Weapons["AK-74M"] = "AK-74M";
    Weapons["ALDA 5.56"] = "ALDA_556";
    Weapons["AR-15.50"] = "AR-1550";
    Weapons["AR33"] = "AR33";
    Weapons["ARX200"] = "ARX200";
    Weapons["AUG A2"] = "AUG_A2";
    Weapons["AUG A3"] = "AUG_A3";
    Weapons["Bailiff 410"] = "Bailiff_410";
    Weapons["BEARING 9"] = "BEARING_9";
    Weapons["BOSG.12.2"] = "BOSG122";
    Weapons["C7E"] = "C7E";
    Weapons["C8-SFW"] = "C8-SFW";
    Weapons["C75 Auto"] = "C75_Auto";
    Weapons["CAMRS"] = "CAMRS";
    Weapons["COMMANDO 9"] = "CAMMANDO_9";
    Weapons["CSRX 300"] = "CSRX300";
    Weapons["D-50"] = "D-50";
    Weapons["DP27"] = "DP27";
    Weapons["F2"] = "F2";
    Weapons["F90"] = "F90";
    Weapons["FMG-9"] = "FMG-9";
    Weapons["FO-12"] = "FO-12";
    Weapons["G8A1"] = "G8A1";
    Weapons["G36C"] = "G36C";
    Weapons["GONNE-6"] = "GONNE-6";
    Weapons["GSH-18"] = "GSH-18";
    Weapons["ITA12L"] = "ITA12L";
    Weapons["ITA12S"] = "ITA12S";
    Weapons["K1A"] = "K1A";
    Weapons["KERATOS .357"] = "KERATOS_357";
    Weapons["L85A2"] = "L85A2";
    Weapons["LFP586"] = "LFP586";
    Weapons["LMG-E"] = "LMG-E";
    Weapons["M4"] = "M4";
    Weapons["M12"] = "M12";
    Weapons["M45 MEUSOC"] = "M45_MEUSOC";
    Weapons["M249"] = "M249";
    Weapons["M249 SAW"] = "M249_SAW";
    Weapons["M590A1"] = "M590A1";
    Weapons["M762"] = "M762";
    Weapons["M870"] = "M870";
    Weapons["M1014"] = "M1014";
    Weapons["Mk 14 EBR"] = "Mk_14_EBR";
    Weapons["MK1 9mm"] = "MK1_9mm";
    Weapons["MK17 CQB"] = "MK17_CQB";
    Weapons["MP5"] = "MP5";
    Weapons["MP5K"] = "MP5K";
    Weapons["MP5SD"] = "MP5SD";
    Weapons["MP7"] = "MP7";
    Weapons["MPX"] = "MPX";
    Weapons["Mx4 Strom"] = "Mx4_Strom";
    Weapons["OTs-03"] = "OTs-03";
    Weapons["P-10C"] = "P-10C";
    Weapons["P9"] = "P9";
    Weapons["P10 RONI"] = "P10_RONI";
    Weapons["P12"] = "P12";
    Weapons["P90"] = "P90";
    Weapons["P226 MK 25"] = "P226_MK_25";
    Weapons["P229"] = "P229";
    Weapons["PARA-308"] = "PARA-308";
    Weapons["PDW9"] = "PDW9";
    Weapons["PMM"] = "PMM";
    Weapons["POF-9"] = "POF-9";
    Weapons["PRB92"] = "PRB92";
    Weapons["Q-929"] = "Q-929";
    Weapons["R4-C"] = "R4-C";
    Weapons["RG15"] = "RG15";
    Weapons["SASG-12"] = "SASG-12";
    Weapons["SC3000K"] = "SC3000K";
    Weapons["SCORPION EVO 3 A1"] = "SCORPION_EVO_3_A1";
    Weapons["SDP 9mm"] = "SDP_9mm";
    Weapons["SG-CQB"] = "SG-CQB";
    Weapons["SIX12"] = "SIX12";
    Weapons["SIX12 SD"] = "SIX12_SD";
    Weapons["SMG-11"] = "SMG-11";
    Weapons["SMG-12"] = "SMG-12";
    Weapons["SPAS-12"] = "SPAS-12";
    Weapons["SPAS-15"] = "SPAS-15";
    Weapons["SPEAR .308"] = "SPEAR_308";
    Weapons["SPSMG9"] = "SPSMG9";
    Weapons["SR-25"] = "SR-25";
    Weapons["SUPER 90"] = "SUPER_90";
    Weapons["SUPER SHORTY"] = "SUPER_SHORTY";
    Weapons["SUPERNOVA"] = "SUPERNOVA";
    Weapons["T-5 SMG"] = "T-5_SMG";
    Weapons["T-95 LSW"] = "T-95_LSW";
    Weapons["TCSG12"] = "TCSG12";
    Weapons["TYPE-89"] = "TYPE-89";
    Weapons["UMP45"] = "UMP45";
    Weapons["USP40"] = "USP40";
    Weapons["UZK50GI"] = "UZK50GI";
    Weapons["V308"] = "V308";
    Weapons["VECTOR .45 ACP"] = "VECTOR_45_ACP";
})(Weapons || (Weapons = {}));
export var Event;
(function (Event) {
    Event["Apocalypse"] = "beret";
    Event["Containment"] = "quarantine";
    Event["DoktorsCurse"] = "doktorcurse";
    Event["FreezeForAll"] = "frosty";
    Event["MuteFleshAndMetal"] = "bot";
    Event["Rengoku"] = "toky";
    Event["Showdown"] = "showdown";
    Event["SnowBrawl"] = "snowball";
    Event["SugarFright"] = "telly";
    Event["TheGrandLarceny"] = "mafia";
    Event["TheTeddyConflict"] = "rimv2";
})(Event || (Event = {}));
export var EsportsTeams;
(function (EsportsTeams) {
    EsportsTeams["00Nation"] = "00NATION";
    EsportsTeams["Astralis"] = "ATRALIS";
    EsportsTeams["Chaos EC"] = "CHAOS_EC";
    EsportsTeams["Cloud9"] = "CLOUD9";
    EsportsTeams["Fav Gaming"] = "FAV_GAMING";
    EsportsTeams["Giants Gaming"] = "GIANTS_GAMING";
    EsportsTeams["Guts Gaming"] = "GUTS_GAMING";
    EsportsTeams["Heroic"] = "HEROIC";
    EsportsTeams["Intz"] = "INTZ";
    EsportsTeams["Invictus Gaming"] = "INVICTUS_GAMING";
    EsportsTeams["Mirage"] = "MIRAGE";
    EsportsTeams["MNM Gaming"] = "MNM_GAMING";
    EsportsTeams["Natus Vincere"] = "NATUS_VINCERE";
    EsportsTeams["Nora Rengo"] = "NORA_RENGO";
    EsportsTeams["Order"] = "ORDER";
    EsportsTeams["Parabellum Esports"] = "PARABELLUM_ESPORTS";
    EsportsTeams["QConfirm"] = "QCONFIRM";
    EsportsTeams["Santos"] = "SANTOS";
    EsportsTeams["Team Empire"] = "TEAM_EMPIRE";
    EsportsTeams["Team Vitality"] = "TEAM_VITALITY";
    EsportsTeams["The Chiefs"] = "THE_CHIEFS";
    EsportsTeams["Xavier Esports"] = "XAVIER_ESPORTS";
    EsportsTeams["XSet"] = "XSET";
})(EsportsTeams || (EsportsTeams = {}));
export var OtherTypes;
(function (OtherTypes) {
    OtherTypes["Battlepass"] = "acq_battlepass";
    OtherTypes["Seasonal"] = "Seasonal";
    OtherTypes["Universal"] = "Universal";
})(OtherTypes || (OtherTypes = {}));
// Sort By options
// Defualt for sell option
export const PurchaseAvailableHighLow = {
    direction: 'DESC',
    field: 'ACTIVE_COUNT',
    orderType: 'Buy',
    paymentItemId: '9ef71262-515b-46e8-b9a8-b6b6ad456c67',
};
export const PurchaseAvailableLowHigh = {
    direction: 'ASC',
    field: 'ACTIVE_COUNT',
    orderType: 'Buy',
    paymentItemId: '9ef71262-515b-46e8-b9a8-b6b6ad456c67',
};
// Default for buy option
export const SaleAvailableHighLow = {
    direction: 'DESC',
    field: 'ACTIVE_COUNT',
    orderType: 'Sell',
    paymentItemId: '9ef71262-515b-46e8-b9a8-b6b6ad456c67',
};
export const SaleAvailableLowHigh = {
    direction: 'ASC',
    field: 'ACTIVE_COUNT',
    orderType: 'Sell',
    paymentItemId: '9ef71262-515b-46e8-b9a8-b6b6ad456c67',
};
export const LastSalePriceHighLow = {
    direction: 'DESC',
    field: 'LAST_TRANSACTION_PRICE',
    paymentItemId: '9ef71262-515b-46e8-b9a8-b6b6ad456c67',
};
export const LastSalePriceLowHigh = {
    direction: 'ASC',
    field: 'LAST_TRANSACTION_PRICE',
    paymentItemId: '9ef71262-515b-46e8-b9a8-b6b6ad456c67',
};
export const ItemNamesDESC = {
    direction: 'DESC',
    field: 'DISPLAY_NAME',
    paymentItemId: '9ef71262-515b-46e8-b9a8-b6b6ad456c67',
};
export const ItemNamesASC = {
    direction: 'ASC',
    field: 'DISPLAY_NAME',
    paymentItemId: '9ef71262-515b-46e8-b9a8-b6b6ad456c67',
};
