// r6statapi/dist/cjs/methods/getServerStatus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetServerStatus = void 0;
const constants_1 = require("../constants");
const auth_1 = require("./auth");
const apiClient_1 = require("./apiClient");
const GetServerStatus = async (platform) => {
    if (platform === '')
        throw new Error(`please check platform: ${platform}`);
    const token = await (0, auth_1.CheckToken)();
    const headers = {
        Authorization: `ubi_v1 t=${token}`,
        'Ubi-AppId': constants_1.UBI_APPID,
        'Content-Type': 'application/json',
    };
    const serverId = constants_1.UBI_SERVER_IDS.find(x => x.id === platform)?.value;
    if (!serverId)
        throw new Error('unable to find serverId');
    const URI = constants_1.UBI_SERVER_STATUS_URI + (0, constants_1.UBI_GETSERVERSTATUS)(serverId);
    const response = await (0, apiClient_1.ApiClient)(URI, headers, 'GET');
    const data = await response;
    const serverStatus = {
        MDM: data[0].MDM,
        SpaceID: data[0].SpaceID,
        Category: data[0].Category,
        Name: data[0].Name,
        platform: data[0].Platform,
        status: data[0].Status,
        maintenance: data[0].Maintenance,
        impactedFeatures: data[0].ImpactedFeatures,
    };
    return serverStatus;
};
exports.GetServerStatus = GetServerStatus;
