// r6statapi/dist/cjs/methods/getUserRank.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetUserRank = void 0;
const apiClient_1 = require("./apiClient");
const auth_1 = require("./auth");
const constants_1 = require("../constants");
const GetUserRank = async (userId, platform) => {
    if (!userId || !platform)
        throw new Error(`Please check userId: ${userId} and platform: ${platform}`);
    const token = await (0, auth_1.CheckToken)();
    const headers = {
        Authorization: `ubi_v1 t=${token}`,
        'Ubi-AppId': constants_1.UBI_APPID,
        'Ubi-SessionId': constants_1.UBI_SESSIONID,
        'Content-Type': 'application/json',
    };
    const platformChange = platform === 'uplay' ? 'pc' : 'console';
    const URI = (0, constants_1.BASE_UBI_URI)(2) + (0, constants_1.UBI_RANKED_URI_V2)(userId, platformChange);
    const response = await (0, apiClient_1.ApiClient)(URI, headers, 'GET');
    const result = await extractValues(await response);
    return result;
};
exports.GetUserRank = GetUserRank;
const extractValues = async (profiles) => {
    const ranks = {};
    profiles.platform_families_full_profiles.forEach((event) => {
        event.board_ids_full_profiles.forEach((profile) => {
            const boardId = profile.board_id;
            const fullProfile = profile.full_profiles[0].profile;
            const seasonStatistics = profile.full_profiles[0].season_statistics;
            const getRank = (0, constants_1.GetRanksById)(fullProfile.rank);
            const rank = {
                profile_board_id: boardId,
                id: fullProfile.board_id,
                max_rank: fullProfile.max_rank,
                max_rank_points: fullProfile.max_rank_points,
                platform_family: fullProfile.platform_family,
                rank: fullProfile.rank,
                rank_points: fullProfile.rank_points,
                rank_name: getRank?.name ?? '',
                season_id: fullProfile.season_id,
                top_rank_position: fullProfile.top_rank_position,
                deaths: seasonStatistics.deaths,
                kills: seasonStatistics.kills,
                abandons: seasonStatistics.match_outcomes.abandons,
                losses: seasonStatistics.match_outcomes.losses,
                wins: seasonStatistics.match_outcomes.wins,
                rankImage: getRank?.image ?? '',
            };
            ranks[boardId] = rank;
        });
    });
    return ranks;
};
