// r6statapi/dist/cjs/methods/searchMarketplace.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Search = void 0;
const auth_1 = require("./auth");
const constants_1 = require("../constants");
const apiClient_1 = require("./apiClient");
const helperFunctions_1 = require("../utils/helperFunctions");
const Search = async (searchQuery, limit) => {
    // end of each of word we need to add * Looks something like this R4-C*
    const query = searchQuery
        .split(' ')
        .map((word) => word + '*')
        .join(' ');
    console.log(query);
    var token = await (0, auth_1.CheckToken)();
    const headers = {
        Authorization: `ubi_v1 t=${token}`,
        'Ubi-AppId': constants_1.UBI_APPID_Marketplace,
        'Ubi-SessionId': constants_1.UBI_SESSIONID_MARKETPLACE,
        'ubi-localecode': constants_1.UBI_LOCALECODE,
        'ubi-regionid': constants_1.UBI_REGIONID_MARKETPLACE,
        'x-platform-appid': constants_1.X_PLATFORM_APPID,
        'Content-Type': 'application/json',
    };
    const body = [
        {
            operationName: 'GetMarketableItems',
            variables: {
                withOwnership: true,
                spaceId: '0d2ae42d-4c27-4cb7-af6c-2099062302bb',
                limit: limit,
                offset: 0,
                filterBy: {
                    text: query,
                },
                sortBy: {
                    field: 'ACTIVE_COUNT',
                    orderType: 'Sell',
                    direction: 'DESC',
                    paymentItemId: '9ef71262-515b-46e8-b9a8-b6b6ad456c67',
                },
            },
            query: constants_1.GraphQL_SearchQuery,
        },
    ];
    const response = await (0, apiClient_1.ApiClient)(constants_1.UBI_MARKETPLACE_URI, headers, 'POST', body);
    const items = await MapSearchResults(response);
    return items;
};
exports.Search = Search;
const MapSearchResults = async (data) => {
    const items = {
        items: [],
    };
    data.forEach((element) => {
        const itemDetails = element.data.game;
        if (itemDetails.marketableItems && itemDetails.marketableItems.nodes.length > 0) {
            itemDetails.marketableItems.nodes.forEach(async (recItem) => {
                const tags = recItem.item.tags;
                const marketData = await (0, helperFunctions_1.MapMarketData)(recItem.marketData);
                const newItem = {
                    id: recItem.item.id,
                    assetUrl: recItem.item.assetUrl,
                    itemId: recItem.item.itemId,
                    name: recItem.item.name,
                    tags: tags,
                    type: recItem.item.type,
                    marktetData: marketData,
                };
                items.items.push(newItem);
            });
        }
    });
    return items;
};
